embassy-rp/src/adc.rs

```diff
@@ -42,24 +46,55 @@ impl<'p> Pin<'p> {
             w.set_pue(pull == Pull::Up);
             w.set_pde(pull == Pull::Down);
         });
-        Self { pin: pin.map_into() }
+        Self(Source::Pin(pin.map_into()))
+    }
+
+    pub fn new_sensor(s: impl Peripheral<P = ADC_TEMP_SENSOR> + 'p) -> Self {
```

Comment:
name should mention "temperature" somehow, just "sensor" doesn't tell what it does.